rclpy/src/rclpy_common/include/rclpy_common/exceptions.hpp

```diff
@@ -65,8 +75,12 @@ class RCLInvalidROSArgsError : public RCLError
 {
 };
 
-class UnknownROSArgsError : public RCLError
+class UnknownROSArgsError : public std::runtime_error
 {
+private:
+  using std::runtime_error::runtime_error;
+
+  friend void throw_if_unparsed_ros_args(py::list, const rcl_arguments_t &);
```

Comment:
@sloretz uh, why the privacy?

Reply:
IIUC `private` with `using std::runtime_error::runtime_error` forces one to use `throw_if_unused_ros_args()` so the exception text is always formatted the same. The access specifier has no effect on the `friend` declaration.

Reply:
I ended up removing the friendship and pybind11 use from rclpy_common (with Windows Debug compatibility in mind) in d224f19